Crashlytics/Crashlytics/Models/FIRCLSInstallIdentifierModel.m

```diff
@@ -98,14 +98,17 @@ - (NSString *)generateInstallationUUID {
 
 #pragma mark Privacy Shield
 
-- (BOOL)regenerateInstallIDIfNeeded {
+- (BOOL)regenerateInstallIDIfNeededWithBlock:(void (^)(NSString *fiid))block {
   BOOL __block didRotate = false;
 
   dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
 
   // This runs Completion async, so wait a reasonable amount of time for it to finish.
   [self.installations
       installationIDWithCompletion:^(NSString *_Nullable currentIID, NSError *_Nullable error) {
+        // Provide the IID to the callback
+        block(currentIID);
```

Comment:
So in this case I don't actually have any way of handling the error besides passing nil to the block and logging a message. Since logging a message is handled in `rotateCrashlyticsInstallUUIDWithIID`, I went with just simply logging what we're dealt. Does that make sense to you?

Reply:
In `rotateCrashlyticsInstallUUIDWithIID`, we stored a fiid hash to UserDefaults `[defaults setObject:currentIIDHash forKey:FIRCLSInstallationIIDHashKey];` and if there is error we return early. Want to make sure we are not changing the fiid if error happen to align with the way we store this fiidHash.

Reply:
Discussed offline, added comments for explanation.